pretyping/cbv.ml

```diff
@@ -628,7 +628,7 @@ and cbv_stack_value info env = function
 
     (* constructor in a Projection -> IOTA *)
     | (CONSTRUCT(((sp,n),u),[||]), APP(args,PROJ(p,_,stk)))
-        when red_set info.reds fMATCH && Projection.unfolded p ->
+        when red_set info.reds fMATCH && red_set info.reds (fPROJ (Projection.repr p)) ->
```

Comment:
shouldn't it be 
```suggestion
        when red_projection p ->
```
?

Reply:
Good question. Using `red_projection` seems indeed the correct entry point. But wasn't the convention that the reduction of a projection was requiring MATCH (and even COFIX for a recursive tuple?); that is, wasn't the convention that red_projection expects some iota to be set?

As for the unfolded, that's indeed better to keep it as for now In the longer term, wouldn't the unfolded/folded status disappear though?

Reply:
> reduction of a projection was requiring MATCH (and even COFIX for a recursive tuple?)

The [refman](https://coq.inria.fr/doc/V8.19.0/refman/proofs/writing-proofs/equality.html?highlight=cbv#grammar-token-reductions) seems silent about this issue. It does not even mention what `match`, `fix`, and `cofix` exactly cover.

In the chapter on inductive types, iota and fix are [mentioned](https://coq.inria.fr/doc/V8.19.0/refman/language/core/inductive.html#destructors).

In the chapter on records, how [reduction of primitive projections](https://coq.inria.fr/doc/V8.19.0/refman/language/core/inductive.html#destructors) is controlled is not explicit.

So, I guess a *necessary* step would already be to decide what we expect to see in the reference manual. For instance, do we add a flag `proj` covering reduction of primitive projections. And do we include the fold-to-unfold step of primitive projections in such `proj` flag or do we consider instead this step as relevant of `delta`, or even of only `delta`? Do we define `iota` to include such `proj` or not? Do we make reduction of match-defined positive projections entirely different from reduction of primitive projections? Etc.

Reply:
sorry for the confusion, I actually meant 
```suggestion
        when red_set info.reds fMATCH && red_projection infos p ->
```
not just red_projection